RampUpAuto/src/test/java/seleautomation/com/RampUpAuto/GitHubSignInPOMDataProviderStartup.java

```diff
@@ -0,0 +1,93 @@
+package seleautomation.com.RampUpAuto;
+
+import java.lang.reflect.Method;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.edge.EdgeDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+
+public class GitHubSignInPOMDataProviderStartup {
+    WebDriver driver = null;
+    GitHubSignInPOM gitHubSignInPOM;
+
+    @BeforeClass
+    @Parameters("browser")
+    public void beforeClass(String browser) {
+        if (browser.equalsIgnoreCase("chrome")) {
+            System.setProperty("webdriver.chrome.driver",
+                    "C:\\Users\\enoor\\Desktop\\Authomation\\EnasRampUpAuto\\ChromeDriver.exe");
+            driver = new ChromeDriver();
+        } else if (browser.equalsIgnoreCase("firefox")) {
+            System.setProperty("webdriver.gecko.driver",
+                    "C:\\Users\\enoor\\Desktop\\Authomation\\EnasRampUpAuto\\geckodriver.exe");
+            driver = new FirefoxDriver();
+        } else if (browser.equalsIgnoreCase("edge")) {
+            System.setProperty("webdriver.edge.driver",
+                    "C:\\Users\\enoor\\Desktop\\Authomation\\EnasRampUpAuto\\msedgedriver.exe");
+            driver = new EdgeDriver();
+        } else {
+            throw new IllegalArgumentException("Invalid browser value provided");
+        }
+
+        gitHubSignInPOM = new GitHubSignInPOM(driver);
+
+
+    }
+
+    @DataProvider(name = "github-signin")
+    public Object[][] githubSigninDb(Method m) {
+
+        switch (m.getName()) {
+            case "validSignIn":
+                return new Object[][] {
+                        { "asaltest19@gmail.com", "password123456789s*" }
+                };
+            case "inValidSignIn":
+                return new Object[][] {
+                        { "asaltest1646449@gmail.com", "password123456789s*" },
+                        { "asaltest19@gmail.com", "password123456789sgg" },
+                        { "", "" }
+                };
+        }
+        return null;
+
+    };
+
+    @BeforeMethod
+    public void beforeTest() {
+        gitHubSignInPOM.openGitHubLoginPage();
+    }
+
+    @Test(dataProvider = "github-signin")
+    public void validSignIn(String email, String password) {
+        gitHubSignInPOM.enterUserName(email);
+        gitHubSignInPOM.enterPassword(password);
+        gitHubSignInPOM.pressLogin();
+        String currentUrl = gitHubSignInPOM.getCurrentUrl();
+        Assert.assertEquals(currentUrl, "https://github.com");
+    }
+
+    @Test(dataProvider = "github-signin")
```

Comment:
Instead of doing 2 separate test functions , you can merge this function with one above as both of them have the same steps    

```
gitHubSignInPOM.enterUserName(email);
        gitHubSignInPOM.enterPassword(password);
        gitHubSignInPOM.pressLogin(); 
```
and to distinguish between the valid and invalid cases , you can add additional parameter as a flag (true or false ) to your data provider as this 
`{ "asaltest1646449@gmail.com", "password123456789s*" , false},`

Reply:
Yes, I agree but I separate the test cases in two methods because they are two different code and I don't want to pass true or false as a third parameter in the @DataProvider especially that java does not have optional parameter directly.